### PR TITLE
Add Potsdam Segmentation

### PR DESCRIPTION
**Pull request by isaaccorley**
This PR adds the [Potsdam](https://www2.isprs.org/commissions/comm2/wg4/benchmark/2d-sem-label-potsdam/) dataset from the [WG II/4 2D Semantic Labeling Contest](https://www2.isprs.org/commissions/comm2/wg4/benchmark/semantic-labeling/).

Notes:
- I used the specified colormap found [here](https://www2.isprs.org/commissions/comm2/wg4/benchmark/semantic-labeling/).
- The labels are provided but I split the dataset by filenames into the train/test sets used in the challenge.
- The ground truth has these light blue blobs which represent the "Low vegetation" class, that is not a bug.
- The images are not georefenced but they are RGBN so I'm reading them using rasterio and it complains that they aren't georeferenced in the tests.

Examples:

image

image

### CODE REVIEW

**Comment by calebrob6** [comment]
Sweet, thanks @isaaccorley! Only thing I saw was adding datamodule to docs and I just did that inline

File: torchgeo/datasets/potsdam.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+from torch.utils.data import DataLoader
+from torchvision.transforms import Compose
+
+from ..datasets.utils import dataset_split, draw_semantic_segmentation_masks
+from .geo import VisionDataset
+from .utils import check_integrity, extract_archive, rgb_to_mask
+
+
+class Potsdam(VisionDataset):
+    """Potsdam dataset.
```
Comment (adamjstewart) [inline torchgeo/datasets/potsdam.py:24]: Is "Potsdam" unique enough to identify this dataset and this dataset only? Or are there other Potsdam datasets (for 3D-semantic segmentation or something like that)? I think we'll want to call this Potsdam2D or maybe something even more specific.

**Review by adamjstewart — COMMENTED** [review]
I assume that if I run the DataModule tests it will expand the zip files and leave a bunch of files that aren't tracked in the repo? Can you expand these zip files and include both raw files and zip files in the PR?

File: torchgeo/datasets/potsdam.py (hunk 1 above)
Comment (isaaccorley) [reply torchgeo/datasets/potsdam.py]: Changed to Potsdam2D

**Comment by isaaccorley** [comment]
Added the files contained in the zip files and changed everything to Potsdam2D.

File: tests/datasets/test_potsdam.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from _pytest.monkeypatch import MonkeyPatch
+
+from torchgeo.datasets import Potsdam2D, Potsdam2DDataModule
+
+
+class TestPotsdam:
```
Comment (adamjstewart) [inline tests/datasets/test_potsdam.py:18]:
```suggestion
class TestPotsdam2D:
```

File: tests/datasets/test_potsdam.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    def test_plot(self, dataset: Potsdam2D) -> None:
+        x = dataset[0].copy()
+        dataset.plot(x, suptitle="Test")
+        dataset.plot(x, show_titles=False)
+        x["prediction"] = x["mask"].clone()
+        dataset.plot(x)
+
+
+class TestPotsdamDataModule:
```
Comment (adamjstewart) [inline tests/datasets/test_potsdam.py:76]:
```suggestion
class TestPotsdam2DDataModule:
```

File: torchgeo/datasets/potsdam.py [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+from ..datasets.utils import dataset_split, draw_semantic_segmentation_masks
+from .geo import VisionDataset
+from .utils import check_integrity, extract_archive, rgb_to_mask
+
+
+class Potsdam2D(VisionDataset):
+    """Potsdam 2D Semantic Segmentation dataset.
+
+    The `Potsdam <https://www2.isprs.org/commissions/comm2/wg4/benchmark/2d-sem-label-potsdam/>`_
+    dataset is a dataset urban semantic segmentation used in the 2D Semantic Labeling
```
Comment (adamjstewart) [inline torchgeo/datasets/potsdam.py:29]:
```suggestion
    dataset is a dataset for urban semantic segmentation used in the 2D Semantic Labeling
```

File: torchgeo/datasets/potsdam.py [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+    Content - Potsdam. This dataset uses the "4_Ortho_RGBIR.zip" and "5_Labels_all.zip"
```
Comment (adamjstewart) [inline torchgeo/datasets/potsdam.py:30]: This sentence is unclear to me, is Content supposed to be Contest? What does the "- Potsdam" at the end mean?

File: torchgeo/datasets/potsdam.py [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+                ax1.set_title("Predictions")
+
+        if suptitle is not None:
+            plt.suptitle(suptitle)
+
+        return fig
+
+
+class Potsdam2DDataModule(pl.LightningDataModule):
+    """LightningDataModule implementation for the Potsdam dataset.
```
Comment (adamjstewart) [inline torchgeo/datasets/potsdam.py:298]:
```suggestion
    """LightningDataModule implementation for the Potsdam2D dataset.
```

File: torchgeo/datasets/potsdam.py (hunk 5 above)
Comment (isaaccorley) [reply torchgeo/datasets/potsdam.py:30]: That's a typo. It should be Contest.

Comment (isaaccorley) [reply torchgeo/datasets/potsdam.py:30]: The contest website splits the 2D Semantic Labeling Contest into 2 sections: Potsdam and Vaihingen. See https://www2.isprs.org/commissions/comm2/wg4/benchmark/2d-sem-label-potsdam/

File: torchgeo/datasets/utils.py [hunk 7]
```diff
@@ -459,3 +460,24 @@ def draw_semantic_segmentation_masks(
...
+
+
+def rgb_to_mask(
+    rgb: np.ndarray, colors: List[Tuple[int, int, int]]  # type: ignore[type-arg]
+) -> np.ndarray:  # type: ignore[type-arg]
+    """Converts an RGB colormap mask to a integer mask.
+
+    Args:
+        rgb: array mask of coded with RGB tuples
+        colors: list of RGB tuples to convert to integer indices
```
Comment (adamjstewart) [inline torchgeo/datasets/utils.py:472]: Need an empty line between Args and Returns sections

File: torchgeo/datasets/potsdam.py [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+from .geo import VisionDataset
+from .utils import check_integrity, extract_archive, rgb_to_mask
+
+
+class Potsdam2D(VisionDataset):
+    """Potsdam 2D Semantic Segmentation dataset.
+
+    The `Potsdam <https://www2.isprs.org/commissions/comm2/wg4/benchmark/2d-sem-label-potsdam/>`_
+    dataset is a dataset for urban semantic segmentation used in the 2D Semantic Labeling
+    Content - Potsdam. This dataset uses the "4_Ortho_RGBIR.zip" and "5_Labels_all.zip"
```
Comment (adamjstewart) [inline torchgeo/datasets/potsdam.py:30]:
```suggestion
    Contest - Potsdam. This dataset uses the "4_Ortho_RGBIR.zip" and "5_Labels_all.zip"
```

**Review by adamjstewart — APPROVED** [review]
